boost_version.txt

```diff
@@ -0,0 +1,4 @@
+# This file is for lock boost version
+# Please update shasum256 in boost_install.sh
+# https://www.boost.org/users/download/
+1.75.0
```

Comment:
having to update two files to match the local boost version is inconvenient.
and the default version in the script doesn't mean only that specific version is supported.

as a better approach, just define two environment variables `boost_version` and `boost_tarball_sha256sum`.